modules/data-entry/src/main/frontend/src/questionnaire/IdentifierQuestion.jsx

```diff
@@ -0,0 +1,117 @@
+//  distributed with this work for additional information
+//  regarding copyright ownership.  The ASF licenses this file
+//  to you under the Apache License, Version 2.0 (the
+//  "License"); you may not use this file except in compliance
+//  with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing,
+//  software distributed under the License is distributed on an
+//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+//  KIND, either express or implied.  See the License for the
+//  specific language governing permissions and limitations
+//  under the License.
+//
+
+import React, { useState } from "react";
+
+import { IconButton, Tooltip } from "@mui/material";
+import withStyles from '@mui/styles/withStyles';
+import ContentCopyIcon from "@mui/icons-material/ContentCopy";
+
+import PropTypes from "prop-types";
+
+import Answer from "./Answer";
+import Question from "./Question";
+import QuestionnaireStyle from "./QuestionnaireStyle";
+import FormattedText from "../components/FormattedText.jsx";
+import AnswerComponentManager from "./AnswerComponentManager";
+
+// Component that renders an identifier question, with optional copy button.
+// Selected answers are placed in a series of <input type="hidden"> tags for
+// submission.
+//
+// Optional props:
+//  text: String containing text to show alongside the identifier
+//  displayMode: "plain", "plain+copy", or undefined denoting how the identifier is presented
+//             to the user. "+copy" adds a button the user can press to copy the value to clipboard.
+//             Defaults to "plain".
+//
+// Sample usage:
+// <NumberQuestion
+//    text="Visit Identifier"
+//    displayMode="plain+copy"
+//    dataType="identifier"
+//    />
+function IdentifierQuestion(props) {
+  const { existingAnswer, classes, pageActive, isEdit, ...rest} = props;
+  const { displayMode } = {...props.questionDefinition };
+  const [ text, setText ] = useState("Copy to Clipboard")
+
+  const initialValue = existingAnswer?.[1]?.value || "";
+  const answer = initialValue === "" ? [] : [["value", initialValue]];
+
+  const handleClick = () => {
+    navigator.clipboard.writeText(initialValue);
+    setText("Copied");
+  }
+
+  const handleClose = () => {
+    setText("Copy to Clipboard");
+  }
+
+  return (
+    <Question
+      disableInstructions
+      preventDefaultView
+      {...props}
+      >
+        { pageActive && <>
+          <FormattedText className={classes.identifierQuestionText}>{initialValue}</FormattedText>
+          { displayMode.endsWith("+copy") &&
+            <Tooltip title={text} onClose={handleClose} className={classes.identifierQuestionButton}>
+              <IconButton onClick={handleClick}>
+                <ContentCopyIcon />
+              </IconButton>
+            </Tooltip>
+          }
+          <div style={{clear: "both"}}></div>
+        </> }
```

Comment:
```suggestion
        { pageActive &&
          ( displayMode.endsWith("+copy") ?
            <Tooltip title={text} onClose={handleClose}>
              <Button
                onClick={handleClick}
                endIcon={<ContentCopyIcon />}
                sx={{padding: 0, textTransform: "none"}}
              >
              { initialValue}
              </Button>
            </Tooltip>
            : <FormattedText on>{ initialValue }</FormattedText>
          )
```

I like how it stands out thanks to the primary color, and was wondering if it would be a good idea to display it as a button and have the tooltip and the copy functionality enabled even when the `displayMode` is `plain` (and simply not show the `Copy` icon in that case).

![image](https://github.com/user-attachments/assets/e3f4c837-c359-454c-9b32-1d5460af98da)

Reply:
Another look option:

![image](https://github.com/user-attachments/assets/0a4a6251-f362-41af-9fd7-67760bfc6054)

done with

```
                <FormattedText sx={{ color: theme => theme.palette.text.primary }}>
                  {"```" + initialValue + "```"}
                </FormattedText>
```

Reply:
![image](https://github.com/user-attachments/assets/f3f1feaf-5dbc-4c80-a695-a0e3c887fc75)